### PR TITLE
[quant] avoid resize calls in observer/fake_quant

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60386 [quant] avoid resize calls in observer/fake_quant**

Summary:
During QAT we sometimes encounter errors with scripted models
`RuntimeError: cannot resize variables that require grad`

For per-tensor cases we don't need to resize some buffers so this PR removes the extra resize ops where applicable
Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29271905](https://our.internmc.facebook.com/intern/diff/D29271905)

### CODE REVIEW

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29271905).

File: torch/quantization/observer.py [hunk 1]
```diff
@@ -533,8 +533,6 @@ def forward(self, x_orig):
             min_val_cur, max_val_cur = torch._aminmax(x)
             min_val = min_val + self.averaging_constant * (min_val_cur - min_val)
             max_val = max_val + self.averaging_constant * (max_val_cur - max_val)
-        self.min_val.resize_(min_val.shape)
```
Comment (vkuzo) [inline torch/quantization/observer.py:536]: can we keep this and gate behind an if statement, like we did for `FakeQuantize?

Comment (supriyar) [reply torch/quantization/observer.py:536]: For this observer we will always have the min/max val have a shape of 1, I believe. We have a separate one - `MovingAveragePerChannelMinMaxObserver` which has different shape.

**Review by vkuzo — APPROVED** [review]
lg if CI passes

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29271905).

**Pull request closed without merging**